src/lib/fstab/tsort.rb

```diff
@@ -0,0 +1,37 @@
+require "tsort"
+
+# Namespace for code related to /etc/fstab
+module Fstab
+  # Simple module implementing topological sorting for fstab entries
+  module TSort
+    extend Yast::Logger
+
+    # Sorts a list of fstab entries by the topological order of the mount points
+    def self.sort(list)
+      # Define the two iterators required by TSort
+      each_node = ->(&b) { list.reverse.each(&b) }
```

Comment:
this looks very ineffective as it for each call need to call reverse. For me better way is `each_node = ->(&b) { list.reverse.method(:each) }